sw/device/lib/dif/dif_hmac.c

```diff
@@ -234,11 +234,15 @@ dif_result_t dif_hmac_finish(const dif_hmac_t *hmac,
   bool done = mmio_region_get_bit32(hmac->base_addr, HMAC_INTR_STATE_REG_OFFSET,
                                     HMAC_INTR_STATE_HMAC_DONE_BIT);
 
+  // Check if fifo_empty is asserted.
```

Comment:
I could be wrong, but this DIF method assumes that the `dif_hmac_process()` was already invoked, to trigger the engine to begin its computation. If the SW / app is not done writing the message to the FIFO (and has not yet triggered the operation) then this method should not be called. I suppose this check is to safeguard against such a user error? 

I am not sure if this is really needed, given DIFs are only used in testing - it's not production code. It is not possible to safeguard against ALL possible user errors.

Reply:
I think this verification is intended to avoid disabling the hmac when it is still processing, and this was working fine when the irqs were not used in the app. But when the irqs are used, the app erase the 'INTR_STATE_REG' before calling this function.
So I think this verification is still needed to erases the 'INTR_STATE_REG' when the irq is not used in the app.

Reply:
Aah, I see - the `isr_testutils_hmac_isr()` invoked in `ottf_external_isr()` ends up already clearing this bit. In that case, the ISR itself is the indication that the HMAC operation is completed. The main flow of execution does not need to call `dif_hmac_finish()`. Instead, invoke `wait_for_interrupt()` to issue a WFI and exit.

Reply:
I have merged the tests hma_enc and hmac_enc_irq.

Reply:
I still don't feel convinced this is needed. If IRQ is enabled, then the receipt of the IRQ (and invoking the standard HMAC IST testutil) replaces this function. I will let @alphan or @timothytrippel comment more and approve if they feel ok with this change. Also, +@eunchan, the HMAC designer.

Reply:
Hi @sriyerg, I came up with another solution here, just remove the whole `else` clause. Since it does not seem fair returning an error when the irq status has been cleared already.